ONSdigital/rm-actionexporter-service
# Delete old records + contact data

# Motivation and Context
Previously all the data from every printfile generated was saved in the actionexporter database in perpetuity.  This data included names and email addresses of respondents.  As part of the GDPR work, we're going to remove the data that is no longer needed.  Every morning at 3am, the /delete-old-records endpoint will be hit and delete all data related to exportFiles that were successfully sent over 90 days ago.  This gives us a reasonable amount of time to replay anything that might need to be replayed, but anything longer then 90 days will almost certainly not be needed again.

Note:  The address data isn't deleted in this PR.  This is more difficult because of the Many to One relationship the addresses have with the action instructions (one address record is used for multiple action instructions) and the documentation isn't super clear how to clear down an address after the last action instruction has finished using it.

# What has changed




# How to test?
- Port forward to your environment
- Run this script [recreate_actionexporter.txt](https://github.com/ONSdigital/rm-actionexporter-service/files/5232758/recreate_actionexporter.txt) against your database to populate some test data
- `curl -X DELETE http://localhost:8141/delete-old-records`, then look at the logs to see one of the exportjobs deleted (and all its associated data)

# Links
